lib/floe/workflow/runner/docker.rb

```diff
@@ -43,9 +43,70 @@ def run!(resource, env = {}, secrets = {})
           secrets_file&.close!
         end
 
+        def run_async!(resource, env = {}, secrets = {})
+          raise ArgumentError, "Invalid resource" unless resource&.start_with?("docker://")
+
+          image = resource.sub("docker://", "")
+
+          params = ["run", :detach]
+          params += [[:net, "host"]] if network == "host"
+          params += env.map { |k, v| [:e, "#{k}=#{v}"] } if env
+
+          secrets_file = nil
+
+          if secrets && !secrets.empty?
+            secrets_file = Tempfile.new
+            secrets_file.write(secrets.to_json)
+            secrets_file.flush
+
+            params << [:e, "SECRETS=/run/secrets"]
+            params << [:v, "#{secrets_file.path}:/run/secrets:z"]
+          end
+
+          params << image
+
+          logger.debug("Running docker: #{AwesomeSpawn.build_command_line("docker", params)}")
+
+          begin
+            container_id = docker!(*params).output
+          rescue
+            secrets_file&.close!
```

Comment:
Should this call `cleanup` instead?  (and perhaps make `cleanup` "safer" in case the container failed to be created?